Bug 513583 - Enable eslint a11y rules

DraciVik opened the pull request:
Fixed jsx-a11y accessibility:

**1. jsx-a11y/control-has-associated-label for files:**

- SecondaryNavBar;
- PushList;
- CompareTable;

**2. jsx-a11y/no-noninteractive-element-interactions for files:**

- FiltersMenu;
- PushActionMenu;
- LegendCard;
- ClassificationGroup;
- TruncatedText;
- Test;

**3. jsx-a11y/no-static-element-interactions for files:**

- BugColumn;
- App;
- PinBoard;
- ActionBar;
- AnnotationsTab;
- SuggestionsListItem;
- ActiveFilters;
- SecondaryNavBar;
- PushList;
- Revision;
- AlertTableRow;
- GraphsContainer;
- LegendCard;
- Metric;
- Test;
===================================

@sarah-clements @camd 
These are the changes I made so the accessibility rules pass. I made a few changes in the css for these files so that there aren't any differences from before these changes.

I noticed that in the codebase a lot of "react-strap" is being used. My opinion is that for a lot of these elements it would be better if I used semantic elements from the 'react-strap' library. I did not want to use it without your approval.

Also as per @sarah-clements request, I rebased my commits so the working history is clean now.

Looking forward to any critiques.
Thank you for your patience!

sarah-clements commented:
> @sarah-clements @camd
> These are the changes I made so the accessibility rules pass. I made a few changes in the css for these files so that there aren't any differences from before these changes.

The requirement was to make changes to 1-2 files so this is certainly an over achievement :)

> I noticed that in the codebase a lot of "react-strap" is being used. My opinion is that for a lot of these elements it would be better if I used semantic elements from the 'react-strap' library. I did not want to use it without your approval.
> 
Yes, this is a good inclination on your part. In most cases, we should be using react strap components. Instead of changing all of the files, you can just change a few though.
> Also as per @sarah-clements request, I rebased my commits so the working history is clean now.
Yes, this is good. Please keep the changes to a separate commit with a title such as "changes based on feedback".

sarah-clements commented:
@DraciVik Please start to finalize your application - you don't need to wait until we approve this pr to do so.

DraciVik commented:
> > @sarah-clements @camd
> > > These are the changes I made so the accessibility rules pass. I made a few changes in the css for these files so that there aren't any differences from before these changes.
> > 
> > The requirement was to make changes to 1-2 files so this is certainly an over achievement :)

I hope it's not a problem. I saw this as an opportunity to get good practice with the 'react-strap' library on a real big project.
> > > I noticed that in the codebase a lot of "react-strap" is being used. My opinion is that for a lot of these elements it would be better if I used semantic elements from the 'react-strap' library. I did not want to use it without your approval.
> >
> > Yes, this is a good inclination on your part. In most cases, we should be using react strap components. Instead of changing all of the files, you can just change a few though.

Thank you for the approval. I changed what I thought would be appropriate with the 'react-strap' library. There was much more that I could do, like, for instance, changing a lot of dropdowns into the 'react-strap' Dropdown component but that would have meant changing code that did not have any 'jsx-a11y' problems to begin with.


> > > Also as per @sarah-clements request, I rebased my commits so the working history is clean now.

> > Yes, this is good. Please keep the changes to a separate commit with a title such as "changes based on feedback".

Did it this way, but between the time of me rebasing my branch and pushing it here a commit was merged on the 'master' branch that threw a conflict for the 'ActionBar.jsx' component which is why I had to amend and force push my commit so the Travis CI tests passed. I bet there are more elegant ways to do this than force push my amended commit so I'm sorry if it's a problem.

DraciVik commented:
@sarah-clements @camd 

I took this opportunity to practice with accessibility for React applications and the 'react-strap' library and I hope it's not a big problem that I did it on all the files that had issues with the jsx-a11y compatibility. 
When these HTML tags were changed to 'react-strap' components the 'react-strap' styles needed to be adjusted so that there aren't any noticeable differences - hence the additional 'classNames'. When styles could not be adjusted as needed with 'classNames' the CSS for the specific component was adjusted.
In places where a group of HTML tags were missing a11y accessibility i used an appropriate 'react-strap' component like 'Dropdown', 'ButtonGroup' in accordance with the context.

In the IRC chat it was mentioned that we should proceed with our Outreachy application without waiting for changes on pull requests to be reviewed and we had time until Friday when you'll be reviewing them.
I am happy to amend any changes in response to your critiques. 
Thank you for your patience! :raised_hands:

DraciVik commented on ui/job-view/details/summary/ActionBar.jsx:
```diff
@@ -495,88 +505,68 @@ class ActionBar extends React.PureComponent {
             )}
-            <li className="dropdown ml-auto">
-              <Button
+            <Dropdown
```
Thank you for your feedback @camd . I changed the `Dropdown` into an `UncontrolledDropdown` and removed all previous code that was necessary for state manipulation.
If anything else needs improvement I'll be happy to oblige.
@sarah-clements

sarah-clements commented:
@DraciVik Since you did a great job with this task, we'd like to assign the bug to you that this pr is for (see https://bugzilla.mozilla.org/show_bug.cgi?id=1513583). 

But before we can merge this you'll need to resolve the file conflicts since there's been changes to the code base since you've worked on this. You'll also need to rename your very first commit (you can do this with rebase using the -i flag) to include the bug number: "Bug 513583 - Enable eslint a11y rules" and update the title of this pull request (at the top of the page). Can you make these changes?

DraciVik commented:
> 
> 
> @DraciVik Since you did a great job with this task, we'd like to assign the bug to you that this pr is for (see https://bugzilla.mozilla.org/show_bug.cgi?id=1513583).
> 
> But before we can merge this you'll need to resolve the file conflicts since there's been changes to the code base since you've worked on this. You'll also need to rename your very first commit (you can do this with rebase using the -i flag) to include the bug number: "Bug 513583 - Enable eslint a11y rules" and update the title of this pull request (at the top of the page). Can you make these changes?

I will definitely do this. I'm traveling right now and as soon as I get settled with a good connection (which shouldn't take more than a day or two) I'll get right on it. @sarah-clements.
I hope the delay is not a problem.

sarah-clements commented:
> > @DraciVik Since you did a great job with this task, we'd like to assign the bug to you that this pr is for (see https://bugzilla.mozilla.org/show_bug.cgi?id=1513583).
> > But before we can merge this you'll need to resolve the file conflicts since there's been changes to the code base since you've worked on this. You'll also need to rename your very first commit (you can do this with rebase using the -i flag) to include the bug number: "Bug 513583 - Enable eslint a11y rules" and update the title of this pull request (at the top of the page). Can you make these changes?
> 
> I will definitely do this. I'm traveling right now and as soon as I get settled with a good connection (which shouldn't take more than a day or two) I'll get right on it. @sarah-clements.
> I hope the delay is not a problem.

No problem at all :) I'll assign that bugzilla bug to you.

DraciVik commented:
> 
> 
> @DraciVik Since you did a great job with this task, we'd like to assign the bug to you that this pr is for (see https://bugzilla.mozilla.org/show_bug.cgi?id=1513583).
> 
> But before we can merge this you'll need to resolve the file conflicts since there's been changes to the code base since you've worked on this. You'll also need to rename your very first commit (you can do this with rebase using the -i flag) to include the bug number: "Bug 513583 - Enable eslint a11y rules" and update the title of this pull request (at the top of the page). Can you make these changes?


@sarah-clements I fixed all the merge conflicts, renamed the first commit in the series and changed the name of the pull request.
After solving the merge conflicts I had a little trouble with ActionBar.jsx not being contained but it's fixed now. 
After that I had tests failing because I did not rebase properly from the master. Took me a while but I think I finally got the hang of it.
If there are any more requirements please let me know. 
**Thank you!**

sarah-clements commented:
Thanks @DraciVik! One last thing, we've discovered that Dropdown Menu's aren't very accessible, so we need to add `tag=a` to the `DropdownItem` components. You can see a more detailed explanation in pr #5725. If you can add those that'd be appreciated.

DraciVik commented:
> 
> 
> Thanks @DraciVik! One last thing, we've discovered that Dropdown Menu's aren't very accessible, so we need to add `tag=a` to the `DropdownItem` components. You can see a more detailed explanation in pr #5725. If you can add those that'd be appreciated.

@sarah-clements Yes, not a problem. I did that the same day but I forgot to mention you so you've probably missed my additional commit. I hope everything is fine now.

sarah-clements commented:
> > Thanks @DraciVik! One last thing, we've discovered that Dropdown Menu's aren't very accessible, so we need to add `tag=a` to the `DropdownItem` components. You can see a more detailed explanation in pr #5725. If you can add those that'd be appreciated.
> 
> @sarah-clements Yes, not a problem. I did that the same day but I forgot to mention you so you've probably missed my additional commit. I hope everything is fine now.

Ah, we don't see notifications when commits are added - only comments, at mentions or review requests. Thanks for doing that but it needs a rebase now due to merge conflicts - sorry!

sarah-clements commented:
I forgot to follow up with this and get it merged but because there are so many merge conflicts, I'm afraid I won't catch all the changes properly. Going to close and assign to someone else.
